request-processor/src/application/core/utils.py

```
@@ -465,6 +465,16 @@ def validate_source(
     existing_entry = _read_existing_source_entry(source_csv_path, source_key_returned)
     if existing_entry:
         source_summary["existing_source_entry"] = existing_entry
+        # A combined endpoint can serve multiple datasets; if this dataset isn't already
+        # listed in pipelines, the existing row needs to be updated with a semicolon-appended value.
+        existing_pipelines = existing_entry.get("pipelines", "")
+        existing_pipeline_list = [p.strip() for p in existing_pipelines.split(";") if p.strip()]
+        if dataset.strip() not in existing_pipeline_list:
+            updated_pipelines = ";".join(existing_pipeline_list + [dataset.strip()])
```

Ruff (0.15.12)

[warning] Consider `[*existing_pipeline_list, dataset.strip()]` instead of concatenation

Replace with `[*existing_pipeline_list, dataset.strip()]`

(RUF005)